dashboard/modules/job/common.py

```diff
@@ -156,7 +158,7 @@ def get_all_jobs(self) -> Dict[str, JobInfo]:
                 self.JOB_DATA_KEY_PREFIX
             ), "Unexpected format for internal_kv key for Job submission"
             job_ids.append(job_id_with_prefix[len(self.JOB_DATA_KEY_PREFIX) :])
-        return {job_id: self.get_info(job_id) for job_id in job_ids}
+        return {job_id: await self.get_info(job_id) for job_id in job_ids}
```

Comment:
This will be blocking calls right? Maybe we should use asyncio.gather here?

Reply:
nice catch